### PR TITLE
End deprecation phase for `switch` cases which skip the declaration of a variable

### PR DESCRIPTION
**Pull request by wilzbach**
This never made it to the deprecation page.

- [x] https://github.com/dlang/dmd/pull/7842

### CODE REVIEW

File: changelog/switch-skip-declaration.dd [hunk 1]
```diff
@@ -0,0 +1,21 @@
+The deprecation for `switch` cases which skip the declaration of a variable has ended
```
Comment (JinShil) [inline changelog/switch-skip-declaration.dd]: Period at the end of this sentence.

File: changelog/switch-skip-declaration.dd [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+Usage of variable which is declared in another `switch` case now results in an error.
```
Comment (JinShil) [inline changelog/switch-skip-declaration.dd]: "Usage of **a** variable..."

File: changelog/switch-skip-declaration.dd [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+            int j;
+        case 2:
+            ++j;
+            break;
+        }
+        default:
+            break;
+    }
+}
+---
```
Comment (JinShil) [inline changelog/switch-skip-declaration.dd]: I think the example should identify where the error is, and provide corrective action.

**Review by JinShil — CHANGES_REQUESTED** [review]

File: changelog/switch-skip-declaration.dd (hunk 1 above)
Comment (wilzbach) [reply changelog/switch-skip-declaration.dd]: This is the title - it's a git commit message like format. Check DAutoTest.

**Comment by mathias-lang-sociomantic** [comment]
There is still https://issues.dlang.org/show_bug.cgi?id=17752 which hasn't been addressed.

File: changelog/switch-skip-declaration.dd (hunk 1 above)
Comment (JinShil) [reply changelog/switch-skip-declaration.dd]: Ah, right.  Sorry.

**Comment by wilzbach** [comment]
Argh I found one more bugs in Bugzilla - [16578](https://issues.dlang.org/show_bug.cgi?id=16578).

**Review by JinShil — APPROVED** [review]

**Comment by JinShil** [comment]
> Argh I found one more bugs in Bugzilla - 16578.

Bug fix submitted here: #7891

**Review by andralex — APPROVED** [review]

**Comment by wilzbach** [comment]
#7891 has been merged, so this should be finally good to go.

**Comment by wilzbach** [comment]
AppVeyor already passed, but the status didn't get reported :/

https://ci.appveyor.com/project/greenify/dmd/build/1.0.3017

image

**Pull request merged**